uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -608,6 +613,21 @@ define([
       return role === roleParam;
     }
 
+    /**
+     * Initialises watchers
+     */
+    function initWatchers () {
+      $rootScope.$watch(function () {
+        return vm.period;
+      }, function () {
+        _loadAbsenceTypes()
+          .then(function () {
+            setInitialAbsenceTypes();
+            $timeout(function () { $rootScope.$emit('LeaveRequestPopup::updateBalance'); }, 0);
```

Comment:
I am not sure that using $timeout for code control is that particular case is a good idea. Why it needs $timeout?

Reply:
I found another bug, after opening the popup, if I directly select a date from next period, the balance calculation was not happening, this was because I changed the flow before, the watcher was getting fired, before the calendar data is loaded. And because of that `from_data_type` was blank, and hence the balance change UI got hidden.

So I realised that the loading of absence period has to be done after the calendar is loaded. So I removed the watcher and added a new event `loadAbsenceTypes`. With this, we dont need the `$timeout`anymore. Updated the description, check for complete details.